### ISSUE TITLE
Resolver Overload

### ISSUE THREAD

**Issue by daftbadger** [ValveSoftware/steam-for-linux#3401]
Hello there ive been trying to use steam for linux of and on now for a while but a persistent problem ive had is that steam "overloads" my resolver.

not sure if this is a bug or if there is some issue with my setup.
but steam keeps trying to open alot of connections at the same time very quickly and my isp does not like this at all.

im using Linux mint 17 64bit cinnamon release

this here below is the message i get redirected to when ive tried downloading my games in steam for a little while. when using windows i do not have this problem at all

---

Overload
You have arrived at this page because your home router or computer is sending too many name queries per second to the 85.11.1.11 resolver.

Reboot your home router, or close the program causing the overload, and then press the link at the bottom of the page.

Normal operation will resume within 5 minutes after that.
Top 10

```
96 content8.steampowered.com
18 content2.steampowered.com
 8 valve25.cs.steampowered.com
 8 content3.steampowered.com
 2 archive.ubuntu.com
 2 www.mentalcast.tk
 1 www.1201beyond.com
 1 r18---sn-5go7dn76.c.android.clients.google.com
 1 steamcloudams.blob.core.windows.net
 1 packages.linuxmint.com
 1 se-ivt.np.community.playstation.net
```

Log
Mon May 7 10:53:57 2012: 77.110.45.172 sent 319 queries in 10 seconds
Sat Jul 13 00:58:41 2013: 77.110.45.172 sent 240 queries in 10 seconds
Sat Jul 13 00:59:01 2013: 77.110.45.172 sent 258 queries in 10 seconds
Sat Jul 13 00:59:21 2013: 77.110.45.172 sent 345 queries in 10 seconds
Sat Jul 13 14:30:51 2013: 77.110.45.172 sent 211 queries in 10 seconds
Sat Jul 13 14:33:01 2013: 77.110.45.172 sent 336 queries in 10 seconds
Sun Jul 14 00:17:11 2013: 77.110.45.172 sent 210 queries in 10 seconds
Fri Jul 26 14:58:51 2013: 77.110.45.172 sent 206 queries in 10 seconds
Thu Aug 8 06:31:31 2013: 77.110.45.172 sent 224 queries in 10 seconds
Thu Aug 8 06:31:41 2013: 77.110.45.172 sent 217 queries in 10 seconds
Thu Sep 5 06:39:51 2013: 77.110.45.172 sent 254 queries in 10 seconds
Thu Sep 5 06:40:01 2013: 77.110.45.172 sent 228 queries in 10 seconds
Thu Sep 5 06:40:11 2013: 77.110.45.172 sent 296 queries in 10 seconds
Thu Sep 5 06:40:41 2013: 77.110.45.172 sent 224 queries in 10 seconds
Mon Sep 30 21:42:31 2013: 77.110.45.172 sent 205 queries in 10 seconds
Sat Nov 23 14:16:34 2013: 77.110.45.172 sent 228 queries in 10 seconds
Sat Nov 23 14:16:54 2013: 77.110.45.172 sent 400 queries in 10 seconds
Sat Nov 23 14:17:04 2013: 77.110.45.172 sent 474 queries in 10 seconds
Sat Nov 23 14:17:14 2013: 77.110.45.172 sent 328 queries in 10 seconds
Sat Nov 23 14:17:24 2013: 77.110.45.172 sent 356 queries in 10 seconds
Sat Nov 23 14:17:34 2013: 77.110.45.172 sent 432 queries in 10 seconds
Sat Nov 23 14:17:44 2013: 77.110.45.172 sent 273 queries in 10 seconds
Sat Nov 23 14:18:14 2013: 77.110.45.172 sent 339 queries in 10 seconds
Sat Nov 23 14:18:44 2013: 77.110.45.172 sent 224 queries in 10 seconds
Sat Nov 23 14:19:04 2013: 77.110.45.172 sent 238 queries in 10 seconds
Sat Nov 23 14:19:14 2013: 77.110.45.172 sent 226 queries in 10 seconds
Sat Nov 23 14:19:24 2013: 77.110.45.172 sent 360 queries in 10 seconds
Sat Nov 23 14:19:34 2013: 77.110.45.172 sent 342 queries in 10 seconds
Sat Nov 23 14:19:44 2013: 77.110.45.172 sent 273 queries in 10 seconds
Sat Nov 23 14:19:54 2013: 77.110.45.172 sent 1148 queries in 10 seconds
Sat Nov 23 14:20:04 2013: 77.110.45.172 sent 552 queries in 10 seconds
Sat Nov 23 16:01:44 2013: 77.110.45.172 sent 276 queries in 10 seconds
Sat Nov 23 16:01:54 2013: 77.110.45.172 sent 272 queries in 10 seconds
Sat Nov 23 16:02:04 2013: 77.110.45.172 sent 373 queries in 10 seconds
Sat Nov 23 16:02:14 2013: 77.110.45.172 sent 295 queries in 10 seconds
Sat Nov 23 16:02:24 2013: 77.110.45.172 sent 220 queries in 10 seconds
Sat Nov 23 16:02:34 2013: 77.110.45.172 sent 275 queries in 10 seconds
Sat Nov 23 16:02:54 2013: 77.110.45.172 sent 268 queries in 10 seconds
Sat Nov 23 16:03:04 2013: 77.110.45.172 sent 240 queries in 10 seconds
Sat Nov 23 16:03:14 2013: 77.110.45.172 sent 230 queries in 10 seconds
Sat Nov 23 16:03:24 2013: 77.110.45.172 sent 281 queries in 10 seconds
Sat Nov 23 16:03:54 2013: 77.110.45.172 sent 315 queries in 10 seconds
Sat Nov 23 16:04:04 2013: 77.110.45.172 sent 439 queries in 10 seconds
Sat Nov 23 16:04:14 2013: 77.110.45.172 sent 224 queries in 10 seconds
Sat Nov 23 16:04:44 2013: 77.110.45.172 sent 340 queries in 10 seconds
Sat Nov 23 16:04:54 2013: 77.110.45.172 sent 450 queries in 10 seconds
Sat Nov 23 16:27:24 2013: 77.110.45.172 sent 325 queries in 10 seconds
Sat Nov 23 16:27:44 2013: 77.110.45.172 sent 238 queries in 10 seconds
Sat Nov 23 16:27:54 2013: 77.110.45.172 sent 414 queries in 10 seconds
Sat Nov 23 16:28:04 2013: 77.110.45.172 sent 299 queries in 10 seconds
Sat Nov 23 16:28:24 2013: 77.110.45.172 sent 264 queries in 10 seconds
Sat Nov 23 16:28:44 2013: 77.110.45.172 sent 351 queries in 10 seconds
Sat Nov 23 16:29:04 2013: 77.110.45.172 sent 598 queries in 10 seconds
Sat Nov 23 16:29:14 2013: 77.110.45.172 sent 305 queries in 10 seconds
Sat Nov 23 16:29:34 2013: 77.110.45.172 sent 806 queries in 10 seconds
Sat Nov 23 16:29:44 2013: 77.110.45.172 sent 533 queries in 10 seconds
Sat Nov 23 16:29:54 2013: 77.110.45.172 sent 299 queries in 10 seconds
Sat Nov 23 16:30:04 2013: 77.110.45.172 sent 370 queries in 10 seconds
Sat Nov 23 18:12:04 2013: 77.110.45.172 sent 285 queries in 10 seconds
Sat Nov 23 18:12:14 2013: 77.110.45.172 sent 352 queries in 10 seconds
Sat Nov 23 18:12:34 2013: 77.110.45.172 sent 292 queries in 10 seconds
Sat Nov 23 18:12:44 2013: 77.110.45.172 sent 1088 queries in 10 seconds
Sat Nov 23 18:12:54 2013: 77.110.45.172 sent 353 queries in 10 seconds
Sat Nov 23 18:13:04 2013: 77.110.45.172 sent 789 queries in 10 seconds
Sat Nov 23 18:14:34 2013: 77.110.45.172 sent 379 queries in 10 seconds
Sat Nov 23 22:41:24 2013: 77.110.45.172 sent 303 queries in 10 seconds
Sat Nov 23 22:41:34 2013: 77.110.45.172 sent 252 queries in 10 seconds
Sat Nov 23 22:41:54 2013: 77.110.45.172 sent 205 queries in 10 seconds
Sat Nov 23 22:43:04 2013: 77.110.45.172 sent 222 queries in 10 seconds
Thu Dec 5 00:15:54 2013: 77.110.45.172 sent 218 queries in 10 seconds
Thu Dec 5 00:16:14 2013: 77.110.45.172 sent 249 queries in 10 seconds
Thu Dec 5 00:16:24 2013: 77.110.45.172 sent 525 queries in 10 seconds
Thu Dec 5 00:34:14 2013: 77.110.45.172 sent 559 queries in 10 seconds
Thu Dec 5 00:34:24 2013: 77.110.45.172 sent 420 queries in 10 seconds
Thu Dec 5 00:36:14 2013: 77.110.45.172 sent 250 queries in 10 seconds
Thu Dec 5 00:36:24 2013: 77.110.45.172 sent 251 queries in 10 seconds
Thu Dec 5 00:36:34 2013: 77.110.45.172 sent 225 queries in 10 seconds
Thu Dec 5 00:36:44 2013: 77.110.45.172 sent 201 queries in 10 seconds
Thu Dec 5 00:37:04 2013: 77.110.45.172 sent 259 queries in 10 seconds
Thu Dec 5 00:37:14 2013: 77.110.45.172 sent 394 queries in 10 seconds
Thu Dec 5 00:37:24 2013: 77.110.45.172 sent 225 queries in 10 seconds
Thu Dec 5 11:39:24 2013: 77.110.45.172 sent 304 queries in 10 seconds
Thu Dec 5 11:39:34 2013: 77.110.45.172 sent 285 queries in 10 seconds
Thu Dec 5 11:39:44 2013: 77.110.45.172 sent 263 queries in 10 seconds
Thu Dec 5 11:39:54 2013: 77.110.45.172 sent 215 queries in 10 seconds
Thu Dec 5 11:40:34 2013: 77.110.45.172 sent 259 queries in 10 seconds
Fri Dec 6 10:49:44 2013: 77.110.45.172 sent 398 queries in 10 seconds
Fri Dec 6 10:49:54 2013: 77.110.45.172 sent 263 queries in 10 seconds
Fri Dec 6 10:50:04 2013: 77.110.45.172 sent 337 queries in 10 seconds
Fri Dec 6 10:50:14 2013: 77.110.45.172 sent 538 queries in 10 seconds
Fri Dec 6 11:05:54 2013: 77.110.45.172 sent 205 queries in 10 seconds
Fri Dec 6 11:06:54 2013: 77.110.45.172 sent 210 queries in 10 seconds
Fri Dec 6 11:13:04 2013: 77.110.45.172 sent 317 queries in 10 seconds
Fri Dec 6 11:13:34 2013: 77.110.45.172 sent 402 queries in 10 seconds
Sat Dec 7 14:02:54 2013: 77.110.45.172 sent 259 queries in 10 seconds
Sat Dec 7 14:03:04 2013: 77.110.45.172 sent 393 queries in 10 seconds
Sat Dec 7 14:03:14 2013: 77.110.45.172 sent 365 queries in 10 seconds
Sat Dec 7 14:03:34 2013: 77.110.45.172 sent 731 queries in 10 seconds
Sat Dec 7 14:04:24 2013: 77.110.45.172 sent 402 queries in 10 seconds
Sat Dec 7 14:04:44 2013: 77.110.45.172 sent 202 queries in 10 seconds
Tue Dec 10 23:59:25 2013: 77.110.45.172 sent 247 queries in 10 seconds
Tue Dec 10 23:59:45 2013: 77.110.45.172 sent 266 queries in 10 seconds
Wed Dec 11 00:08:35 2013: 77.110.45.172 sent 239 queries in 10 seconds
Wed Dec 11 00:09:05 2013: 77.110.45.172 sent 247 queries in 10 seconds
Wed Dec 11 00:11:45 2013: 77.110.45.172 sent 254 queries in 10 seconds
Wed Dec 11 00:13:15 2013: 77.110.45.172 sent 236 queries in 10 seconds
Wed Dec 11 00:16:25 2013: 77.110.45.172 sent 225 queries in 10 seconds
Wed Dec 11 00:16:45 2013: 77.110.45.172 sent 259 queries in 10 seconds
Wed Dec 11 00:20:05 2013: 77.110.45.172 sent 207 queries in 10 seconds
Wed Dec 11 00:21:35 2013: 77.110.45.172 sent 771 queries in 10 seconds
Wed Dec 11 18:11:15 2013: 77.110.45.172 sent 315 queries in 10 seconds
Wed Dec 11 18:12:15 2013: 77.110.45.172 sent 284 queries in 10 seconds
Wed Dec 11 18:16:35 2013: 77.110.45.172 sent 218 queries in 10 seconds
Wed Dec 11 18:17:15 2013: 77.110.45.172 sent 206 queries in 10 seconds
Wed Dec 11 18:17:25 2013: 77.110.45.172 sent 203 queries in 10 seconds
Wed Dec 11 18:18:05 2013: 77.110.45.172 sent 279 queries in 10 seconds
Sat Dec 14 02:40:55 2013: 77.110.45.172 sent 296 queries in 10 seconds
Sat Dec 14 02:45:45 2013: 77.110.45.172 sent 389 queries in 10 seconds
Sat Dec 14 02:46:05 2013: 77.110.45.172 sent 653 queries in 10 seconds
Sat Dec 14 02:46:15 2013: 77.110.45.172 sent 307 queries in 10 seconds
Sat Dec 14 03:15:15 2013: 77.110.45.172 sent 365 queries in 10 seconds
Sat Dec 14 03:15:45 2013: 77.110.45.172 sent 243 queries in 10 seconds
Sat Dec 14 03:15:55 2013: 77.110.45.172 sent 249 queries in 10 seconds
Sat Dec 14 03:16:05 2013: 77.110.45.172 sent 415 queries in 10 seconds
Sat Dec 14 03:16:15 2013: 77.110.45.172 sent 535 queries in 10 seconds
Sat Dec 14 03:16:25 2013: 77.110.45.172 sent 475 queries in 10 seconds
Sat Dec 14 03:16:55 2013: 77.110.45.172 sent 331 queries in 10 seconds
Sat Dec 14 03:17:25 2013: 77.110.45.172 sent 225 queries in 10 seconds
Sat Dec 14 03:17:55 2013: 77.110.45.172 sent 310 queries in 10 seconds
Sat Dec 14 03:18:05 2013: 77.110.45.172 sent 220 queries in 10 seconds
Sat Dec 14 03:18:25 2013: 77.110.45.172 sent 386 queries in 10 seconds
Sat Dec 14 03:19:05 2013: 77.110.45.172 sent 593 queries in 10 seconds
Sat Dec 14 03:19:15 2013: 77.110.45.172 sent 246 queries in 10 seconds
Sat Dec 14 03:19:25 2013: 77.110.45.172 sent 220 queries in 10 seconds
Sat Dec 14 03:19:45 2013: 77.110.45.172 sent 237 queries in 10 seconds
Sat Dec 14 03:19:55 2013: 77.110.45.172 sent 343 queries in 10 seconds
Sat Dec 14 04:02:55 2013: 77.110.45.172 sent 547 queries in 10 seconds
Sat Dec 14 04:03:05 2013: 77.110.45.172 sent 201 queries in 10 seconds
Sat Dec 14 04:03:45 2013: 77.110.45.172 sent 230 queries in 10 seconds
Sat Dec 14 04:03:55 2013: 77.110.45.172 sent 222 queries in 10 seconds
Sat Dec 14 04:04:05 2013: 77.110.45.172 sent 260 queries in 10 seconds
Sat Dec 14 04:04:15 2013: 77.110.45.172 sent 361 queries in 10 seconds
Sat Dec 14 04:04:35 2013: 77.110.45.172 sent 350 queries in 10 seconds
Sat Dec 14 04:04:45 2013: 77.110.45.172 sent 459 queries in 10 seconds
Sat Dec 14 04:04:55 2013: 77.110.45.172 sent 527 queries in 10 seconds
Sat Dec 14 04:05:05 2013: 77.110.45.172 sent 355 queries in 10 seconds
Sat Dec 14 14:04:15 2013: 77.110.45.172 sent 777 queries in 10 seconds
Sat Dec 14 14:04:25 2013: 77.110.45.172 sent 742 queries in 10 seconds
Sat Dec 14 14:04:35 2013: 77.110.45.172 sent 305 queries in 10 seconds
Sat Dec 14 14:04:45 2013: 77.110.45.172 sent 321 queries in 10 seconds
Sat Dec 14 14:05:25 2013: 77.110.45.172 sent 218 queries in 10 seconds
Sat Dec 14 14:05:55 2013: 77.110.45.172 sent 399 queries in 10 seconds
Sat Dec 14 14:06:05 2013: 77.110.45.172 sent 252 queries in 10 seconds
Sat Dec 14 14:06:15 2013: 77.110.45.172 sent 225 queries in 10 seconds
Sat Dec 14 14:06:25 2013: 77.110.45.172 sent 353 queries in 10 seconds
Sat Dec 14 14:06:45 2013: 77.110.45.172 sent 292 queries in 10 seconds
Sat Dec 14 14:06:55 2013: 77.110.45.172 sent 224 queries in 10 seconds
Sat Dec 14 14:07:15 2013: 77.110.45.172 sent 321 queries in 10 seconds
Sat Dec 14 14:34:15 2013: 77.110.45.172 sent 498 queries in 10 seconds
Sat Dec 14 14:34:25 2013: 77.110.45.172 sent 267 queries in 10 seconds
Sat Dec 14 14:34:45 2013: 77.110.45.172 sent 206 queries in 10 seconds
Sat Dec 14 14:36:05 2013: 77.110.45.172 sent 371 queries in 10 seconds
Sat Dec 14 14:36:45 2013: 77.110.45.172 sent 290 queries in 10 seconds
Sat Dec 14 14:37:05 2013: 77.110.45.172 sent 334 queries in 10 seconds
Sat Dec 14 14:38:35 2013: 77.110.45.172 sent 292 queries in 10 seconds
Sat Dec 14 14:39:15 2013: 77.110.45.172 sent 529 queries in 10 seconds
Sat Dec 14 14:39:25 2013: 77.110.45.172 sent 308 queries in 10 seconds
Sat Dec 14 18:01:35 2013: 77.110.45.172 sent 202 queries in 10 seconds
Sat Dec 14 18:01:45 2013: 77.110.45.172 sent 364 queries in 10 seconds
Sat Dec 14 18:01:55 2013: 77.110.45.172 sent 354 queries in 10 seconds
Sat Dec 14 18:02:05 2013: 77.110.45.172 sent 211 queries in 10 seconds
Sat Dec 14 18:02:15 2013: 77.110.45.172 sent 241 queries in 10 seconds
Sat Dec 14 18:02:25 2013: 77.110.45.172 sent 207 queries in 10 seconds
Sat Dec 14 18:02:35 2013: 77.110.45.172 sent 461 queries in 10 seconds
Sat Dec 14 18:02:55 2013: 77.110.45.172 sent 254 queries in 10 seconds
Sat Dec 14 18:03:45 2013: 77.110.45.172 sent 341 queries in 10 seconds
Sat Dec 14 18:04:05 2013: 77.110.45.172 sent 222 queries in 10 seconds
Sat Dec 14 18:04:15 2013: 77.110.45.172 sent 361 queries in 10 seconds
Sat Dec 14 18:04:35 2013: 77.110.45.172 sent 200 queries in 10 seconds
Sat Dec 14 21:11:55 2013: 77.110.45.172 sent 202 queries in 10 seconds
Sat Dec 14 21:12:55 2013: 77.110.45.172 sent 206 queries in 10 seconds
Sat Dec 14 21:13:05 2013: 77.110.45.172 sent 354 queries in 10 seconds
Fri Dec 20 02:01:55 2013: 77.110.45.172 sent 218 queries in 10 seconds
Fri Dec 20 02:02:45 2013: 77.110.45.172 sent 339 queries in 10 seconds
Mon Jan 13 23:21:31 2014: 77.110.45.172 sent 304 queries in 10 seconds
Mon Jan 13 23:27:21 2014: 77.110.45.172 sent 428 queries in 10 seconds
Mon Jan 13 23:27:41 2014: 77.110.45.172 sent 445 queries in 10 seconds
Mon Jan 13 23:28:01 2014: 77.110.45.172 sent 214 queries in 10 seconds
Mon Jan 13 23:28:41 2014: 77.110.45.172 sent 243 queries in 10 seconds
Mon Jan 13 23:29:01 2014: 77.110.45.172 sent 403 queries in 10 seconds
Mon Jan 13 23:29:21 2014: 77.110.45.172 sent 447 queries in 10 seconds
Mon Jan 13 23:29:41 2014: 77.110.45.172 sent 202 queries in 10 seconds
Mon Jan 13 23:30:01 2014: 77.110.45.172 sent 264 queries in 10 seconds
Fri Jan 17 10:12:41 2014: 77.110.45.172 sent 522 queries in 10 seconds
Fri Jan 17 10:13:21 2014: 77.110.45.172 sent 519 queries in 10 seconds
Fri Jan 17 10:15:11 2014: 77.110.45.172 sent 232 queries in 10 seconds
Fri Jan 17 10:15:21 2014: 77.110.45.172 sent 396 queries in 10 seconds
Fri Jan 17 10:15:41 2014: 77.110.45.172 sent 409 queries in 10 seconds
Fri Jan 17 10:15:51 2014: 77.110.45.172 sent 292 queries in 10 seconds
Fri Jan 17 10:16:31 2014: 77.110.45.172 sent 408 queries in 10 seconds
Fri Jan 17 10:16:51 2014: 77.110.45.172 sent 324 queries in 10 seconds
Fri Jan 17 10:17:01 2014: 77.110.45.172 sent 302 queries in 10 seconds
Fri Jan 17 10:17:11 2014: 77.110.45.172 sent 426 queries in 10 seconds
Fri Jan 17 10:17:31 2014: 77.110.45.172 sent 270 queries in 10 seconds
Fri Jan 17 10:17:41 2014: 77.110.45.172 sent 207 queries in 10 seconds
Fri Jan 17 10:17:51 2014: 77.110.45.172 sent 309 queries in 10 seconds
Fri Jan 17 10:18:01 2014: 77.110.45.172 sent 284 queries in 10 seconds
Fri Jan 17 10:18:11 2014: 77.110.45.172 sent 231 queries in 10 seconds
Fri Jan 17 10:18:21 2014: 77.110.45.172 sent 342 queries in 10 seconds
Fri Jan 17 10:18:31 2014: 77.110.45.172 sent 230 queries in 10 seconds
Fri Jan 17 10:18:51 2014: 77.110.45.172 sent 275 queries in 10 seconds
Fri Jan 17 10:19:01 2014: 77.110.45.172 sent 318 queries in 10 seconds
Fri Jan 17 10:19:11 2014: 77.110.45.172 sent 505 queries in 10 seconds
Fri Jan 17 10:19:31 2014: 77.110.45.172 sent 373 queries in 10 seconds
Fri Jan 17 10:19:51 2014: 77.110.45.172 sent 372 queries in 10 seconds
Fri Jan 17 10:20:01 2014: 77.110.45.172 sent 283 queries in 10 seconds
Wed Feb 5 18:17:31 2014: 77.110.45.172 sent 204 queries in 10 seconds
Wed Feb 5 18:17:41 2014: 77.110.45.172 sent 328 queries in 10 seconds
Wed Feb 5 18:20:31 2014: 77.110.45.172 sent 219 queries in 10 seconds
Wed Feb 5 18:20:51 2014: 77.110.45.172 sent 210 queries in 10 seconds
Wed Feb 5 18:21:51 2014: 77.110.45.172 sent 207 queries in 10 seconds
Wed Feb 5 18:22:01 2014: 77.110.45.172 sent 210 queries in 10 seconds
Wed Feb 5 18:23:51 2014: 77.110.45.172 sent 238 queries in 10 seconds
Wed Feb 5 18:24:21 2014: 77.110.45.172 sent 218 queries in 10 seconds
Sat Apr 26 15:23:03 2014: 77.110.45.172 sent 321 queries in 10 seconds
Sat Jul 12 17:05:00 2014: 77.110.45.172 sent 785 queries in 10 seconds
Sat Jul 12 17:07:00 2014: 77.110.45.172 sent 284 queries in 10 seconds
Sat Jul 12 17:07:30 2014: 77.110.45.172 sent 229 queries in 10 seconds
Sat Jul 12 17:08:00 2014: 77.110.45.172 sent 308 queries in 10 seconds
Sat Jul 12 17:08:50 2014: 77.110.45.172 sent 329 queries in 10 seconds
Sat Jul 12 17:09:10 2014: 77.110.45.172 sent 291 queries in 10 seconds
Sat Jul 12 17:10:40 2014: 77.110.45.172 sent 320 queries in 10 seconds
Sat Jul 12 17:11:20 2014: 77.110.45.172 sent 201 queries in 10 seconds
Sat Jul 12 17:11:30 2014: 77.110.45.172 sent 230 queries in 10 seconds
Sat Jul 12 17:21:00 2014: 77.110.45.172 sent 342 queries in 10 seconds
Sat Jul 12 17:21:30 2014: 77.110.45.172 sent 215 queries in 10 seconds
Sat Jul 12 17:21:40 2014: 77.110.45.172 sent 351 queries in 10 seconds
Sat Jul 12 17:21:50 2014: 77.110.45.172 sent 280 queries in 10 seconds
Sat Jul 12 17:22:20 2014: 77.110.45.172 sent 285 queries in 10 seconds
Sat Jul 12 17:22:30 2014: 77.110.45.172 sent 385 queries in 10 seconds
Sat Jul 12 17:22:40 2014: 77.110.45.172 sent 318 queries in 10 seconds
Sat Jul 12 17:22:50 2014: 77.110.45.172 sent 634 queries in 10 seconds
Sat Jul 12 17:23:00 2014: 77.110.45.172 sent 398 queries in 10 seconds
Sat Jul 12 17:24:20 2014: 77.110.45.172 sent 249 queries in 10 seconds
Sat Jul 12 17:24:40 2014: 77.110.45.172 sent 623 queries in 10 seconds
OK
http://www.google.com/search?q=resolver+overload

**Comment by bartekrychlicki** [comment]
The problem still exists, installing dnsmasq under ubuntu does help/

**Comment by jhoppmann** [comment]
It also works in Debian quite well. Install dnsmasq, add the line "nameserver 127.0.0.1" as first entry in /etc/resolv.conf, then restart the dnsmasq service. Did the trick for me, finally downloading with my full bandwith :)

**Comment by fabiengb** [comment]
Hi,
I am having a weird DNS issue when downloading games too.
I have done wireshark sniffing, i have noticed there are around 80 DNS requests per seconds for the address valve236.cs.steampowered.com.

Download starts, but at some point, Steam will try to download a chunk by connecting to localhost instead of steam server's IP. I actually have a hosts file with a blacklist of ads websites to filter annoyings ads when surfing, BUT this blacklist does not include any *.steampowered.com.

Anyway, when restoring a 'normal' hosts file, the download is running just fine.

So it seems like something weird is going on here:

DNS	88	Standard query 0xde30  A valve236.cs.steampowered.com
DNS	104	Standard query response 0xde30  A 212.73.205.178
127.0.0.1	127.0.0.1	HTTP	461	GET /depot/218063/chunk/cddcf9ab4850fbbcf56b0c1799677a3ac0d0be7e HTTP/1.1 

Why 127.0.0.1 as 212.73.205.178 was returned by DNS ?

**Comment by keithjr** [comment]
This is still an issue in the latest version of Steam (1427176184).

**Comment by systemsoverload** [comment]
+1 - still a massive issue. Was actually locking up my crappy cable modem with DNS flood.

**Comment by atriusofbricia** [comment]
Continues to be an issue. Will it ever be resolved?

**Comment by codeape** [comment]
Indeed this bug pretty much sank my home network. I have 1Gb fiber in to my house and it took me a while until I connected that every time I started Steam on Linux all my internet connected devices got very limited internet access. My home router simply did not cope with that DNS flood. So I guess this is one of the reasons we do not see more than 1% Linux users on Steam. 

Please fix this.

**Comment by MrSchism** [comment]
While this does remain a problem, the suggested solution of using dnsmasq has worked for me.  I use it on the router level (dd-wrt) to allow for better performance across all of my devices.

@bartekrychlicki did you ever get dnsmasq to work properly for you?

**Comment by codeape** [comment]
I installed dnsmasq and it worked for me. However that is a workaround. This kind of bugs can be tricky to diagnose and could actually end up flooding external DNS services.

**Comment by Plaque-fcc** [comment]
I found that my router tricks that with announcing down DNS servers it received from ISP instead of relaying DNS queries, so it doesn't get a DoS.

-- 
Простите за краткость, создано в K-9 Mail.

**Comment by Bednar87** [comment]
Affects me as well. Installed dnsmasq on Ubuntu to resolve the issue.

**Comment by Tebro** [comment]
Wow, this is quite the bug. I am on arch and resolved it by installing dnsmasq and having NetworkManager manage dnsmasq for DNS caching.

**Comment by TierraDelFuego** [comment]
I have a dns caching server running on the LAN (not dnsmasq) and filed this issue:
https://github.com/ValveSoftware/SteamOS/issues/365

By chance I happened to see this issue and I'm posting here because all these issues may be related.
They may be related because of:
warning: Using 'gethostbyname' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking

Do some searching on that error message above and take a dive down the rabbit hole.
[:rabbit::cyclone:](http://stackoverflow.com/a/15165424)

tl;dr if you want a portable application that won't encounter name lookup problems you have to roll your own solution and maybe Valve have rolled their own or maybe not. I am guessing these problems are due to this despite what Valve have implemented.

People stating that they resolved the issue by installing dnsmasq should really be saying they worked-around the issue by installing dnsmasq. Because the issue is still there.

**Comment by tsmike** [comment]
Bug is still there.

>add the line "nameserver 127.0.0.1" as first entry in /etc/resolv.conf, then restart the dnsmasq service
This is unnecessary, since dnsmasq will ignore localhost in resolv.conf. (If you check your logs you should see something like "ignoring nameserver 127.0.0.1 - local interface")

**Comment by MrSchism** [comment]
> People stating that they resolved the issue by installing dnsmasq should really be saying they worked-around the issue by installing dnsmasq. Because the issue is still there.

Absolutely correct.  It's still dumping the request into memory; it just has to resolve.  While the hit on RAM and processor is negligible, it's still there.  I'm not 100% certain, but I believe the packets still generate (which produces a negative effect on the network stack's performance), they just don't leave the source... unless you use DNSMasq on a network device like I do.

**Comment by silins** [comment]
I can confirm that issue on Debian testing.

The dnsmasq  hack relives symptoms, but this is no fix! This is a major bug that has been left unfixed for *YEARS*! It is absolutely unacceptable, Valve. I am a paying customer and your product is basically unusable for me. (I was having 6 BYTES/s downloads!)

This only reinforces my commitment to buy games only on GOG, if available there, no matter the price difference.

**Comment by bartekbrak** [comment]
@silins As if GOG had no bugs, read their sh installers.

**Comment by silins** [comment]
@bartekbrak You are right, GOG is not perfect. But the important thing is exactly that I indeed have red their sh installers, because I can! With GOG I freely download my bought content for archiving and use in perpetuity. With steam I have to use their always on-line DRM client and when they have bugs, change of management/policy/whatever it is all over.

Anyway, sorry for starting a rant here, this is not the place.

**Comment by tmdag** [comment]
dnsmasq hack does not work for me

Still getting crazy low speeds

[http://imgur.com/fSL0euC](url)

**Comment by matthew-cline** [comment]
If you're using NetworkManager, then you [have to change the NetworkManager configuration to use dnsmasq](https://wiki.archlinux.org/index.php/dnsmasq#NetworkManager) rather than using dnsmasq directly.

**Comment by skini151** [comment]
It's almost July 2016 and this issue is not fixed, how sad Valve like Youtube forgot, who made them who they are today. Content creators and users. It's a shame I smell M$ in them,but that's another story.
So fix it already or go open-source and let people fix it if you're to lazy.

**Comment by adrogon** [comment]
The dnsmasq workaround did not work for me either.

Using the Windows 7 or 10 Steam client, I'm constantly over 20 MBps.
Using Ubuntu 14.04, I barely reach 200 kB/s.

Upping this in hope it gets attention: the Linux client is useless in those conditions.

Please tell us what kind of feedback we can provide to help with the resolution.

**Comment by kztd** [comment]
i did sudo apt-get install dnsmasq, but no difference.  Is there a way to check if it is correct service(s) are running / not running?

**Comment by justxux** [comment]
You need to restart your dnsmasq service : sudo service dnsmasq restart

**Comment by kztd** [comment]
thanx justxux, do I need to do that every-time I reboot?

**Comment by justxux** [comment]
kztd, No.

**Comment by bagage** [comment]
Still requiring the `dnsmasq` work-around.

**Comment by austen0** [comment]
Just a quick note, I believe that after installing `dnsmasq` NetworkManager needs to be restarted as well: `sudo service network-manager restart`. I got no improvement until doing this.

**Comment by eknoes** [comment]
``dnsmasq`` Workaround worked for me (Arch Linux), but Download speed is still slower then usual!

**Comment by bagage** [comment]
Any chance to have an official opinion on this? (cc @kisak-valve @gdrewb-valve).

**Comment by kisak-valve** [comment]
Can someone retest this with the current steam beta client (2017-03-08)?

**Comment by hit0ri** [comment]
@kisak-valve, here are my test results:

### Steam stable
Built: Jan 19 2017, at 01:23:20
Steam API: v017
Steam package versions: 1484790260

**DNS requests:**
75-263 requests average
**Download speed:**
18.2 Mbps average
18.5 Mbps max


### Steam Beta
Built: Mar 9 2017, at 07:33:27
Steam API: v017
Steam package versions: 1489075449

**DNS requests:**
8-11 requests average
**Download speed:**
24.4 Mbps average
26.8 Mbps max

*NOTE: DNS stats taken from my router (mikrotik 951Ui-2nD)*

**Comment by kisak-valve** [comment]
Thanks for retesting @hit0ri, hopefully others are seeing the same and we can close this issue out after the next stable client update.

**Comment by kisak-valve** [comment]
The [steam stable client](http://store.steampowered.com/news/28060/) has been bumped. Closing based on the recent test feedback.

**Comment by bagage** [comment]
I can confirm that the situation has improved significantly - thanks for those who have taking care of that 2-years old bug :).

**Comment by fuglede** [comment]
This issue still affects the March 22 2017 build; and dnsmasq still saves the day.

**Comment by kisak-valve** [comment]
Hello @fuglede, can you please include some data that can be pondered.

**Comment by fuglede** [comment]
Tried three different mirrors over the course of 20 minutes, and after a brief period of being able to download, I would get stuck at some ~50kB/s. After having tried with the third mirror, I set dnsmasq, and the moment I changed the resolver (without restarting Steam), I would jump up to the expected ~2MB/s and remain there since.

**Comment by kisak-valve** [comment]
Since this is resolved for many of the affected users, let's continue investigating this on #4975.

**Comment by LQ747** [comment]
Steps to fix issue on Arch Linux:
sudo nano /etc/dnsmasq.config(uncomment filterwin2k)
sudo nano /etc/resolv.config ( add this line nameserver 127.0.0.1)

**Comment by sam0x17** [comment]
Just installing `dnsmasq` fixed this issue for me

**Comment by Bereim** [comment]
@LQ747 's solution was a definite improvement but did not resolve the issue.

**Comment by j0e10** [comment]
> Steps to fix issue on Arch Linux:
> 
>     1. sudo nano /etc/dnsmasq.config
> 
>     2. search for 'listen-address' and set it to 'listen-address=127.0.0.1' (remove the # if it's there).
> 
>     3. systemctl restart dnsmasq.service

Bro you are the goat for this, went from downloading @ 50 mbps to 450 mbps

**Comment by Ailadir** [comment]
> > Steps to fix issue on Arch Linux:
> > ```
> > 1. sudo nano /etc/dnsmasq.config
> > 
> > 2. search for 'listen-address' and set it to 'listen-address=127.0.0.1' (remove the # if it's there).
> > 
> > 3. systemctl restart dnsmasq.service
> > ```
> 
> Bro you are the goat for this, went from downloading @ 50 mbps to 450 mbps
For me its works too,but only after i restarted networkmanager : 
`sudo systemctl restart NetworkManager 
`

**Comment by RabidWallaby** [comment]
> Steps to fix issue on Arch Linux:
> 
> 1. sudo nano /etc/dnsmasq.config
> 2. search for 'listen-address' and set it to 'listen-address=127.0.0.1' (remove the # if it's there).
> 3. systemctl restart dnsmasq.service

I followed these steps, as well as [Bereim](https://github.com/Bereim)'s suggestion of pointing NetworkManager at dnsmasq.  I went from 25Mbps (still slow for me) to 6-7Mbps.  So, it made things MUCH worse.

Suggestions?

**Comment by grimwaldr** [comment]
> @LQ747 's solution was a definite improvement but did not resolve the issue.
> 
> Edit: I had to configure Network Manager to use dnsmasq.
> 
> Additional Steps for Network Manager 4. sudo nano /etc/NetworkManager/conf.d/DNS.Config 5. Add the following to the file: [main] dns=dnsmasq 6. systemctl restart NetworkManager.service
> 
> Steps taken from: https://devopstales.github.io/linux/networkmanagger-dnsmasq/

While this did improve my speed - I am sitting on a 3.5gb/s connection speed and I find I seem to cap out at about 1.0gb/s. I can easily get close to max speed on steam via Windows. Sorta sucks that this is still a bug in 2024.

**Comment by RabidWallaby** [comment]
I found a fix, but it meant disabling HTTP2 inside Steam.  Immediately back
to full speeds.

https://www.reddit.com/r/linux_gaming/comments/16e1l4h/slow_steam_downloads_try_this/

**Comment by YoinkerBoinker** [comment]
This isn'T really fixing the issue. it'S still slower compared to windows.

**Comment by zgaduj** [comment]
I resolve this issue but I don't know how :D
```
sudo systemctl disable systemd-resolved
sudo systemctl stop systemd-resolved
sudo systemctl restart dnsmasq
sudo service dnsmasq restart
sudo service dnsmasq start
sudo service dnsmasq status
# there failed start dnsmasq
# dnsmasq is disabled
sudo systemctl start systemd-resolved

```

and *steam_dev.cfg*:
```
@nClientDownloadEnableHTTP2PlatformLinux 0
@cMaxInitialDownloadSources 25
@fDownloadRateImprovementToAddAnotherConnection 0.5
@cMaxSimultaneousDownloads 20
```

extra config in *sysctl.conf*:
```
net.ipv6.conf.all.disable_ipv6 = 1
net.ipv6.conf.default.disable_ipv6 = 1
net.ipv6.conf.lo.disable_ipv6 = 1

net.core.somaxconn = 2048
net.ipv4.tcp_max_syn_backlog = 2048
net.core.netdev_max_backlog = 10000

net.core.rmem_max = 16777216
net.core.wmem_max = 16777216
net.ipv4.tcp_rmem = 4096 87380 16777216
net.ipv4.tcp_wmem = 4096 65536 16777216
```
then: `sudo sysctl -p`

After this all I have unlocked fully speed :smile:

**Comment by ghost** [comment]
For me removing dnsmasq-base and installing dnsmasq did the trick (on Ubuntu).

**Comment by gsagardia** [comment]
I used Cyberpunk 2077 as the test dummy, since it has a pretty large size.

For those that didn't know like me that stumble upon this, you can open the steam console with `steam steam://open/console` in terminal.

According to steam console `@cMaxSimultaneousDownloads` is no longer a command.

For context I have about **2.2 Gbps up** and about **1.8Gbps down**. I was at about **12 to 13 MBps (Bytes)** before trying the 3 steam console commands. **Theoretical Max should be around 250MBps.**

```
@nClientDownloadEnableHTTP2PlatformLinux 0
@cMaxInitialDownloadSources 25
@fDownloadRateImprovementToAddAnotherConnection 0.5
```

**_I had to stop the download, uninstall, and restart the download after each command for changes to take effect. I ran each for about 3 minutes to see how it changed overtime._** 

- After disabling HTTP2, Real ttime: 100MBps, Peak was 115 MBps
- after running `@cMaxInitialDownloadSources 15` (25 was worse for me), Real time: ~95MBps, Peak: 115MBps
- after running `@fDownloadRateImprovementToAddAnotherConnection 0.5` Speeds fluctuated greatly. Lowest 91MBps, Peak 162MBps, Average ~ 115 MBps

I let the download run longest on the last one, since it's peak was much higher. About 5 minutes

re-ran another to see if results were the same with all 3 changes.
- Avg ~ 120 MBps with much less fluctuations. 

Exited steam client to see if the changes would reset. Ran another fresh download. No console.
- Average ~ 110 MBps
- Edited sysctl.conf with @zgaduj 'fixes' but could never really go over sustain more than 120 MBps on average. It didn't seem to have much or any effect.

Creating the steam_dev.cfg yielded the greatest results, with about a 10x increase in speeds. 
**I wish I could get closer to 250MBps but I'll take a 10x boost.**

**Comment by gsagardia** [comment]
Left my PC on, and went to sleep. Woke up and speeds have reverted back to 12MBps. restarting steam or PC does nothing.
steam_dev.cfg does not appear to work after all.

**Comment by Reverrb** [comment]
There is no way they don't know about this issue
